### PR TITLE
Add initial Ubuntu 20.04 CIS Profiles

### PR DESCRIPTION
**Pull request by cipherboy**
#### Description:

This pull request adds initial Ubuntu 20.04 CIS profiles (at all four combinations of applicability) based on the existing profile mechanism. We use profile inheritance to minimize the size of profiles after CIS Level 1 server. This is based on work originally on pull request #6416 and supersedes it. Included are completed profiles minus rules not yet in upstream from our internal fork; those may be added at later time.

Additionally, this pull request adds missing CIS references and Ubuntu prodtypes to rules used in these profiles so they could build. This is done using automated tooling previously upstreamed (or attempted to be, see the failed autorefer pull request in #6908). Note that sorting of references was not done, as that PR isn't yet merged (#6882) either.

Once this change set is merged, in the immediate future, other changes to existing rules will be upstreamed. Additionally, depending on how this goes, @dodys is working on a STIG profile, which we might want to upstream as well. 

**Note**: At this point in time, the profiles themselves are completed, but testing of all rules &c for Ubuntu applicability/functionality is not yet done and obviously not all rule changes have yet been upstreamed (though, numerous changes have been made internally). If you'd like us to mark it `documentation_complete: false` as a result, let us know. 

#### Rationale:

Maintaining prodtypes and references downstream, even with utilities to help, has resulted in significant merge conflicts. Contributing the profiles to the community and upstreaming the prodtypes/reference changes alone will help with that effort significantly.

---

```
    Co-Authored-By: Brian Turek <brian.turek@gmail.com>
    Co-Authored-By: Gabriel Becker <ggasparb@redhat.com>
    Co-Authored-By: Richard Maciel Costa <richard.maciel.costa@canonical.com>
    Signed-off-by: Alexander Scheel <alex.scheel@canonical.com>
```

/cc @richardmaciel-canonical @dodys

### CODE REVIEW

File: products/ubuntu2004/profiles/cis_level1_server.profile [hunk 1]
```diff
@@ -0,0 +1,891 @@
...
+    - kernel_module_udf_disabled
+
+    #### 1.1.1.7 Ensure mounting of FAT filesystems is limited (Manual)
+    # Skip due to being Level 2
+
+    ### 1.1.2 Ensure /tmp is configured (Automated)
+    - partition_for_tmp
+
+    ### 1.1.3 Ensure nodev option set on /tmp partition (Automated)
+    # Needs rule
```
Comment (cipherboy) [inline products/ubuntu2004/profiles/cis_level1_server.profile:38]: Just as an FYI, these and others in this subsection are ones I was talking about w.r.t. `_optional` form; we can introduce them upstream in a future PR.

**Review by yuumasato — APPROVED** [review]
Changes look good to me, but I haven't checked the Ubuntu CIS references.

@cipherboy I trust that somebody else has reviewed them for you.

**Comment by cipherboy** [comment]
Rebased to fix merge conflicts.

**Comment by cipherboy** [comment]
Thanks for the merge, @yuumasato :-)

**Pull request merged**